tst/server/Event.test.ts

```diff
@@ -71,3 +71,38 @@ describe("addEvent() tests", () => {
         expect(EventSchema.create).toHaveBeenCalledTimes(1);
     });
 });
+
+describe("deleteEvent() tests", () => {
+    test("valid event", async () => {
+        const mockEvent = {
+            name: "test",
+        };
+
+        EventSchema.findById = jest.fn().mockResolvedValue(mockEvent);
+        await expect(getEvent("602734007d7de15fae321153")).resolves.toEqual(mockEvent);
+    });
+
+    test("invalid parameters", async () => {
+        expect.assertions(1);
+        await expect(getEvent("")).rejects.toThrowError("Invalid id");
+    });
+
+    test("no event with that id", async () => {
+        expect.assertions(1);
+        // mongoose returns undefined when a query results in no results
+        const mockEvent = undefined;
+
+        //mock EventSchema.findById to return mockEvent, which will then throw an error
+        EventSchema.findById = jest.fn().mockResolvedValue(mockEvent);
+        await expect(getEvent("602734007d7de15fae321152")).rejects.toThrowError("Event does not exist");
+    });
+
+    test("event deleted", async () => {
+        const mockEvent = {
+            name: "test",
+        };
+
+        EventSchema.findByIdAndDelete = jest.fn().mockResolvedValue(mockEvent);
+        await expect(getEvent("602734007d7de15fae321152")).rejects.toThrowError("Event does not exist");
```

Comment:
What this line does is it runs `getEvent()` with a certain ID and expects to reject the promise returned by the function. What you need to do is run `deleteEvent()` and make sure that `EventSchema.findByIdAndDelete` was ran with certain parameters. 

You can do this by mocking the implementation of `findByIdAndDelete` with the following: `EventSchema.findByIdAndDelete = jest.fn().mockImplementation(async (event: Event) => event);`. Then you can run the function and have checks to make sure things worked correctly:
`await addEvent(mockEvent);`
`expect(EventSchema.findByIdAndDelete).lastCalledWith(mockEvent);`
`expect(EventSchema.findByIdAndDelete).toHaveBeenCalledTimes(1);`

Reply:
That makes a lot of sense, thank you!